### PR TITLE
Added abstract Application class to keep Cli\Console and Mvc\Application DRY

### PR DESCRIPTION
**Pull request by SidRoberts**

### CODE REVIEW

**Comment by Green-Cat** [comment]
:+1:

File: phalcon/application.zep [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		}
+	}
+
+	/**
+	 * Sets the events manager
+	 */
+	public function setEventsManager(<ManagerInterface> eventsManager)
+	{
+		let this->_eventsManager = eventsManager;
+	}
```
Comment (sergeyklay) [inline phalcon/application.zep]: can you please add `return this;` here?

File: phalcon/application.zep [hunk 2]
```diff
@@ -0,0 +1,150 @@
+
+/*
+ +------------------------------------------------------------------------+
+ | Phalcon Framework                                                      |
+ +------------------------------------------------------------------------+
+ | Copyright (c) 2011-2015 Phalcon Team (http://www.phalconphp.com)       |
```
Comment (sergeyklay) [inline phalcon/application.zep]: 2011-2016

File: phalcon/application.zep [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	 *		'frontend' => array(
+	 *			'className' => 'Multiple\Frontend\Module',
+	 *			'path' => '../apps/frontend/Module.php'
+	 *		),
+	 *		'backend' => array(
+	 *			'className' => 'Multiple\Backend\Module',
+	 *			'path' => '../apps/backend/Module.php'
+	 *		)
+	 *	));
+	 *</code>
```
Comment (sergeyklay) [inline phalcon/application.zep]: can you please remove extra slashes here? also use short array syntax. we are in the 21st century :)

File: phalcon/application.zep [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	 */
+	public function registerModules(array modules, boolean merge = false) -> <Application>
+	{
+		var registeredModules;
+
+		if merge === false {
+			let this->_modules = modules;
+		} else {
```
Comment (sergeyklay) [inline phalcon/application.zep]: I think you can remove `else` here by adding `return`

File: phalcon/application.zep [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		return this->_modules;
+	}
+
+	/**
+	 * Gets the module definition registered in the application via module name
+	 *
+	 * @param string name
+	 * @return array|object
+	 */
+	public function getModule(string! name)
```
Comment (sergeyklay) [inline phalcon/application.zep]: remove phpdoc please

File: phalcon/application.zep (hunk 3 above)
Comment (SidRoberts) [inline phalcon/application.zep]: Extra slashes?

Comment (sergeyklay) [inline phalcon/application.zep]: Sorry, extra spaces. Just use:
```php
/**
 * <code>
 * $this->registerModules(...
 * </code>
 */
```
Without spaces / tabs

**Comment by sergeyklay** [comment]
Looks good to me. @andresgutierrez ?

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**